org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -442,9 +442,9 @@ public void accommodatePhysicalActionsIfPresent() {
             var actions = Iterables.filter(IteratorExtensions.toIterable(resource.getAllContents()), Action.class);
             for (Action action : actions) {
                 if (Objects.equal(action.getOrigin(), ActionOrigin.PHYSICAL)) {
-                    // If the unthreaded runtime is requested, use the threaded runtime instead
+                    // If the unthreaded runtime is not requested by the user, use the threaded runtime instead
                     // because it is the only one currently capable of handling asynchronous events.
-                    if (!targetConfig.threading) {
+                    if (!targetConfig.threading && !targetConfig.setByUser.contains(TargetProperty.THREADING)) {
```

Comment:
I'm a little confused about this change. What does it do? I seems entirely unrelated to the PR.